sdcm/teardown_validators/rackaware.py

```diff
@@ -0,0 +1,138 @@
+import logging
+import time
+
+from sdcm.sct_events import Severity
+from sdcm.sct_events.teardown_validators import ValidatorEvent
+from sdcm.teardown_validators.base import TeardownValidator
+
+LOGGER = logging.getLogger(__name__)
+
+
+class RackawareValidator(TeardownValidator):  # pylint: disable=too-few-public-methods
+    """
+    Rack-aware validation is enabled under these conditions:
+    - A rack-aware policy is configured.
+    - A single loader is employed (due to Scylla rack metric limitation).
+    - The cluster is deployed across multiple availability zones or a simulated rack environment or in multiple regions.
+
+    The validation aims to confirm that no coordination traffic is directed to availability zones where loaders are absent
+    """
+    validator_name = 'rackaware'
+
+    # This variable defines the allowed percentage variation of system CQL reads directed to availability zones without loaders.
+    # These reads, such as those performed by nemeses or non-rack-aware test code, may be routed this way.
+    EXPECTED_NON_SYSTEM_READS_PERC = 10
+
+    @property
+    def count_loaders(self):
+        """
+        this function operates at a cluster level, effectively checking how many loaders the entire cluster
+        (regardless of tenant separation) has
+        """
+        loaders = sum(len(loaders.nodes) for loaders in self.tester.loaders_multitenant)
+        LOGGER.debug("Loaders count: %s", loaders)
+        return loaders
+
+    def validate(self):
+        if not self.tester.is_rack_aware_policy:
+            LOGGER.info("No workloads were running under the rack-aware policy.")
+            return
+
+        validation_passed = set()
+        # The case: multi-region cluster
+        if self.tester.db_cluster.test_config.MULTI_REGION:
+            count_loaders = self.count_loaders
+            if count_loaders == 1:
+                # The case: there is one loader located in one of regions (of multi-region cluster)
+                non_rack_db_nodes = self.get_db_non_rack_coordless_nodes(dc_idx=self.tester.loaders.nodes[0].dc_idx,
+                                                                         rack=self.tester.loaders.nodes[0].node_rack,
+                                                                         clusters_list=self.tester.db_clusters_multitenant)
+                rack_db_nodes = self.get_db_rack_nodes(non_rack_nodes=non_rack_db_nodes,
+                                                       clusters_list=self.tester.db_clusters_multitenant)
+                validation_passed.add(self.one_region_validate(non_rack_db_nodes=non_rack_db_nodes,
+                                                               rack_db_nodes=rack_db_nodes))
+            else:
+                # The case: there are a few loaders in different regions
+                for count, loaders in enumerate(self.tester.loaders_multitenant):
```

Comment:
for me it looks like mixing multitenancy with multi_region
if multitenant was meant, should be `index` (index of tenant in test) instead of `count`
and also `if self.tester.db_cluster.test_config.MULTI_REGION:` refers to one cluster - not multitenant (IIRC, `self.tester.db_cluster` refers to first tenant)

Reply:
changed, not use multitenance